### ISSUE TITLE
Training dataset

### ISSUE THREAD

**Issue by reorder-cv** [AlexeyAB/darknet#2642]
Dear AlexeyAB,
Thanks for support on yolov3 model in darknet framework. Its really helpful for people like me. I am currently working on training custom objects   
I have few queries.
**Dataset -**
I trained the object according to the point mentioned in How to improve object detection
 a custom object with positive and negative images in the ratio of 1:1 and 1:0.5 and 1:0 
1. The prediction result shows that the .weight files without negative images shows better accuracy than with negative images. is it really necessary to have negative images in dataset  ?

2. If number of classes are 1000 and if i want to added one more class with the previous one all I have to do is start the training for all 1001 classes from beginning. This consumes the time in days / in weeks? Is there any alternative solution available for this problem to reduce the training time?

**Comment by AlexeyAB** [comment]
@reorder-cv Hi,

1. Do you check mAP on Training or Validation dataset? Try to check mAp on Validation dataset. If this reduces accuracy, do not use negative samples.

2. You can do partial if your previously trained model up to the first [yolo] layer ( 94 layers for yolov3.cfg):
`./darknet partial cfg/yolov3_old.cfg yolov3_old.weights yolov3_old.conv.94 94`

And then train new model by using OLD_images + NEW_images and by using this pre-trained weights
`./darknet detector train data/obj.data cfg/yolov3_new.cfg yolov3_old.conv.94`

**Comment by kenrubiooo** [comment]
Hi, @AlexeyAB! 

Question about the number of layers @reorder-cv will use for re-training. Why is it that the number of layers should be 94? Thank you!

**Comment by AlexeyAB** [comment]
@kenrubiooo Hi,

Because if you changed the number of classes, then you changed the number of filters in this conv-layer: https://github.com/AlexeyAB/darknet/blob/f5fd086407ef3caacc98672c3b8ad8c0e2e0491c/cfg/yolov3.cfg#L603

So all remaining weights will be shifted, so you can't use remaining weights after 94 layer.

**Comment by kenrubiooo** [comment]
Sorry, I cannot quite understand the shifting of weights. Can you explain it more? Thank you!
